src/logics/ArithLogic.cc

```diff
@@ -269,18 +269,17 @@ void ArithLogic::termSort(vec<PTRef>& v) const
     sort(v, LessThan_deepPTRef(*this));
 }
 
-std::string ArithLogic::protectName(std::string const & name, SRef retSort, bool isNullary) const {
-    assert(not name.empty());
-    // TODO: We cannot use isNumConst here since it takes a SymRef.  We cannot take a SymRef because this
-    // function needs to be callable from TemplateFunction, which does not have a SymRef.
-    if (isSortNum(retSort) and isNullary and std::all_of(name.begin(), name.end(), [](unsigned char c) { return std::isdigit(c); })) {
-        return name; // Is a number, no escaping
-    } else if (hasQuotableChars(name) or std::isdigit(name[0]) or isReservedWord(name)) {
-        std::stringstream ss;
-        ss << '|' << name << '|';
-        return ss.str();
+bool ArithLogic::isAmbiguousNullarySymbolName(std::string_view name) const {
```

Comment:
I am thinking that this override can also be removed, maybe?
In the `Logic::disambiguateName`, it only considers `nullary` symbols as candidates for disambiguation. What if it restricted that further to `nullary not interpreted`?

Reply:
I think that should work. I'll commit a suggestion.